src/Psy/Command/SandboxCommand.php

```diff
@@ -0,0 +1,269 @@
+    protected $current;
+
+    /**
+     * @var InputInterface
+     */
+    protected $input;
+
+    /**
+     * @var OutputInterface
+     */
+    protected $output;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('sandbox')
+            ->setDefinition(array(
+                new InputArgument(
+                    'action',
+                    InputArgument::REQUIRED,
+                    'Action to perform over the sandboxes add|list|switch|delete|exit'
+                ),
+                new InputArgument(
+                    'name',
+                    InputArgument::OPTIONAL,
+                    'Optional name for the sandbox'
+                ),
+            ))
+            ->setDescription('Manages sandboxed environments.')
+            ->setHelp(
+                <<<HELP
+sandbox add [name] creates a encapsulated sandbox with an optional name, if not provided will autogenerate a name.
+sandbox list Will show a list of the currently generated sandboxes
+sandbox which Will show the current sandbox
+sandbox switch name Switches to sandbox by name,
+sandbox delete [name] Removes a sandbox by name, or the current if name is not provided.
+sandbox exit Erases all sandboxes and sets the normal state.
+HELP
+            );
+
+        $this->restoreFolder = exec('pwd');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->input = $input;
+        $this->output = $output;
+
+        $action = $this->input->getArgument('action');
+        $name = $this->input->getArgument('name');
+        if (!in_array($action, $this->allowedActions)) {
+            throw new \InvalidArgumentException("Action $action is not allowed.");
+        }
+
+        $this->performAction($action, $name);
+    }
+
+    /**
+     * @param $action
+     * @param $name
+     */
+    protected function performAction($action, $name)
+    {
+        switch ($action) {
+            case self::ACTION_ADD:
+                $identifier = $this->createSandbox($name);
+                $this->switchSandbox($identifier);
+                break;
+            case self::ACTION_LIST:
+                $this->listSandboxes();
+                break;
+            case self::ACTION_WHICH:
+                $this->showCurrentSandbox();
+                break;
+            case self::ACTION_SWITCH:
+                $this->switchSandbox($name);
+                break;
+            case self::ACTION_DELETE:
+                $this->removeSandbox($name);
+                break;
+            case self::ACTION_EXIT:
+                $this->restoreState();
+                break;
+        }
+    }
+
+    /**
+     * @param $name
+     * @return string
+     */
+    protected function getSandboxPath($name)
+    {
+        return sys_get_temp_dir() . DIRECTORY_SEPARATOR . self::SANDBOX_FOLDER . DIRECTORY_SEPARATOR . $name;
```

Comment:
Let's use Configuration::getRuntimeDir instead of `sys_get_temp_dir` for the base directory.